bandwidth-file-spec.txt

```diff
@@ -479,95 +621,260 @@
+
+    "desc_obs_bw_mean=" Int
 
       [Zero or one time.]
 
       The descriptor observed bandwidth mean for this relay in bytes per
       second.
 
+      This KeyValue was added in version 1.2.0 of this specification.
+
+    "relay_recent_measurements_excluded_error_count=" Int
+
+      [Zero or one time.]
+
+      The number of recent relay measurements which were excluded because
+      the measurement was attempted, but failed. Measurements are recent
+      if they are in the last data_period days (5 by default).
+
+      (See the note in section 1.4, version 1.4.0, about excluded relays.)
+
+      This KeyValue was added in version 1.4.0 of this specification.
+
+    "relay_recent_measurements_excluded_near_count=" Int
+
+      [Zero or one time.]
+
+      When all of a relay's recent successful measurements were performed in
+      a period of time that was too short (by default 1 day), the relay is
+      excluded. This KeyValue contains the number of recent successful
+      measurements for the relay that were ignored for this reason.
+
+      (See the note in section 1.4, version 1.4.0, about excluded relays.)
+
+      This KeyValue was added in version 1.4.0 of this specification.
+
+    "recent_measurements_excluded_old_count=" Int
+
+      [Zero or one time.]
+
+      The number of successful measurements for this relay that are too old
+      (more than 5 days, by default).
+
+      (See the note in section 1.4, version 1.4.0, about excluded relays.)
+
+      This KeyValue was added in version 1.4.0 of this specification.
+
+    "recent_measurements_excluded_few_count=" Int
+
+      [Zero or one time.]
+
+      The number of successful measurements for this relay that were ignored
+      because the relay did not have enough successful measurements (fewer
+      than 2, by default).
+
+      (See the note in section 1.4, version 1.4.0, about excluded relays.)
+
+      This KeyValue was added in version 1.4.0 of this specification.
+
+    "vote=" bool
+
+      [Zero or one time.]
+
+      If the value is 0, Tor directory authorities SHOULD ignore the relay's
+      entry in the bandwidth file. They SHOULD vote for the relay the same
+      way they would vote for a relay that is not present in the file.
+
+      This MAY be the case when this relay was not successfully measured but
+      the is included in the Bandwidth File, to diagnose why they were not
+      measured.
+
+      If the value is 1 or the KeyValue is not present, Tor directory
+      authorities MUST use the relay's bw value in any votes for that relay.
+
+      Because Tor versions 0.3.4.11, 0.3.5.8, 0.4.0.2-alpha, and earlier
+      ignore this KeyValue, generator implementations MUST set "bw=1"
+      for a relay when Tor directory authorities SHOULD ignore its relay
+      line. Using the minimum bw value allows authorities that do not
+      understand "vote=0" or "unmeasured=1" to produce sensible votes
```

Comment:
sensible might be a bit abstract?

Reply:
I'll say "reasonable votes for unmeasured relays".

Reply:
I rewrote this part in c8cf8db.